integration_test.go

```diff
@@ -2515,3 +2515,122 @@ func TestCacheRegions(t *testing.T) {
 	}
 
 }
+
+// TestNewTableFromSnapshot tests the ability to create a snapshot from a table,
+// and then use this snapshot to create a new, different table from the table the
+// snapshot was created from. This is different from restoring the snapshot to the
+// table it was created from.
+func TestNewTableFromSnapshot(t *testing.T) {
+	var (
+		key          = t.Name() + "_Get"
+		snapshotName = "snapshot-" + table
+	)
+
+	c := gohbase.NewClient(*host, gohbase.RpcQueueSize(1))
+	defer c.Close()
+	// Insert some data into the main test table.
+	if err := insertKeyValue(c, key, "cf", []byte{1}); err != nil {
+		t.Fatal(err)
+	}
+
+	ac := gohbase.NewAdminClient(*host)
+	// Create snapshot from the main test table.
+	sn, err := hrpc.NewSnapshot(context.Background(), snapshotName, table)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = ac.CreateSnapshot(sn); err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		err = ac.DeleteSnapshot(sn)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Delete the data from main test table after taking snapshot.
+	if err = deleteKeyValue(c, key, "cf", []byte{1}); err != nil {
+		t.Fatal(err)
+	}
+	// Confirm data has been deleted.
+	gMain, err := hrpc.NewGetStr(context.Background(), table, key)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r, err := c.Get(gMain)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(r.Cells) != 0 {
+		t.Fatalf("expected no cells in table %s key %s", table, key)
+	}
+
+	// Name of new table to restore snapshot data to.
+	// The new table is not created at this time.
+	// HBase will do this when trying to restore a snapshot to a table that does not already exist.
+	// If the snapshot table doesn't exist, as in this case, HBase will clone the snapshot to a new
+	// table.
+	tableNew := fmt.Sprintf("gohbase_test_%d_%s", time.Now().UnixNano(), t.Name())
+
+	// Update snapshot to new table:
+	sn.SetTable(tableNew)
```

Comment:
Do we really a new method? It feels like to me doing `sn, err := hrpc.NewSnapshot(context.Background(), snapshotName, tableNew); [...] ac.RestoreSnapshot(sn)` is enough (i.e just create a new snapshot object with the same name but a new table name)

Reply:
Done!